src/cli/rustup_mode.rs

```diff
@@ -1153,6 +1154,29 @@ fn toolchain_remove(cfg: &Cfg, m: &ArgMatches<'_>) -> Result<()> {
     Ok(())
 }
 
+fn override_by_toolchain(cfg: &Cfg, m: &ArgMatches<'_>) -> Result<()> {
+    let toolchain = m.value_of("+toolchain").expect("");
+    let toolchain = cfg.get_toolchain(&toolchain[1..], false)?;
+
+    let status = if !toolchain.is_custom() {
+        Some(toolchain.install_from_dist_if_not_installed()?)
+    } else if !toolchain.exists() {
+        return Err(ErrorKind::ToolchainNotInstalled(toolchain.name().to_string()).into());
+    } else {
+        None
+    };
+
+    let path = utils::current_dir()?;
+    toolchain.make_override(&path)?;
+
+    if let Some(status) = status {
+        println!();
+        common::show_channel_update(cfg, toolchain.name(), Ok(status))?;
+    }
```

Comment:
One difficulty here is that this is far too noisy if you already have a toolchain installed.  consider:

```console
somniloquy% rustup +nightly which rustc                                                   
info: using existing install for 'nightly-x86_64-unknown-linux-gnu'
info: override toolchain for '/home/dsilvers' set to 'nightly-x86_64-unknown-linux-gnu'

  nightly-x86_64-unknown-linux-gnu unchanged - rustc 1.40.0-nightly (032a53a06 2019-10-03)

/home/dsilvers/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/bin/rustc
somniloquy%                                                                               
```

In addition it appears to have added an override which is definitely not desirable.  We need to come up with a way to not have this override in a persistent fashion or we cannot have this patch.

Reply:
Will making the changes in https://github.com/rust-lang/rustup.rs/pull/2031/files#diff-a11274be6e834e18a2579dad1b37ce8dR1162 help?

Reply:
That link doesn't quite seem to show me anything I understand as a fix for this.  The problem is the presence of the `make_override()` I think.